capability/capability_linux.go

```diff
@@ -336,21 +341,16 @@ func (c *capsV3) Apply(kind CapType) (err error) {
 		var data [2]capData
 		err = capget(&c.hdr, &data[0])
 		if err != nil {
-			return
+			return err
 		}
 		if (1<<uint(CAP_SETPCAP))&data[0].effective != 0 {
 			for i := Cap(0); i <= last; i++ {
 				if c.Get(BOUNDING, i) {
 					continue
 				}
 				err = prctl(syscall.PR_CAPBSET_DROP, uintptr(i), 0, 0, 0)
-				if err != nil {
-					// Ignore EINVAL since the capability may not be supported in this system.
-					if err == syscall.EINVAL { //nolint:errorlint // Errors from syscall are bare.
-						err = nil
-						continue
-					}
-					return
+				if err != nil && !ignorableError(err) {
```

Comment:
Also, with this change, in case `kind` only has `BOUNDS` and the kernel returns `EINVAL` for the last PR_CAPBSET_DROP, it will be returned, while the code before your changes returns `nil` in the same case.

Reply:
I don't think so. `ignorableError` is not the main reason of this refactor, removing the named return param is the main refactor, `err` is now a local val, not the return name. It will help to avoid causing other issues like the original issue of `ambient` implementation.